flink-runtime/src/main/java/org/apache/flink/runtime/failure/FailureEnricherUtils.java

```diff
@@ -0,0 +1,227 @@
+import org.apache.flink.core.plugin.PluginUtils;
+import org.apache.flink.util.FlinkRuntimeException;
+import org.apache.flink.util.concurrent.FutureUtils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/** Utils class for loading and running pluggable failure enrichers. */
+public class FailureEnricherUtils {
+
+    private static final Logger LOG = LoggerFactory.getLogger(FailureEnricherUtils.class);
+    // regex pattern to split the defined failure enrichers
+    private static final Pattern enricherListPattern = Pattern.compile("\\s*,\\s*");
+    static final String MERGE_EXCEPTION_MSG =
+            "Trying to merge a label with a duplicate key %s. This is a bug that should be reported,"
+                    + " because Flink shouldn't allow registering enrichers with the same output.";
+
+    public static Collection<FailureEnricher> getFailureEnrichers(
+            final Configuration configuration) {
+        final PluginManager pluginManager =
+                PluginUtils.createPluginManagerFromRootFolder(configuration);
+        return getFailureEnrichers(configuration, pluginManager);
+    }
+
+    /**
+     * Returns a set of validated FailureEnrichers for a given configuration, job ID, job name,
+     * metric group, and plugin manager.
+     *
+     * @param configuration the configuration for the job
+     * @param pluginManager the PluginManager for the job
+     * @return a collection of validated FailureEnrichers
+     * @throws IllegalArgumentException if any of the FailureEnrichers are invalid
+     */
+    static Collection<FailureEnricher> getFailureEnrichers(
+            final Configuration configuration, final PluginManager pluginManager) {
+        Set<String> includedEnrichers = getIncludedFailureEnrichers(configuration);
+        //  When empty, NO enrichers will be started.
+        if (includedEnrichers.isEmpty()) {
+            return Collections.emptySet();
+        }
+        final Iterator<FailureEnricherFactory> factoryIterator =
+                pluginManager.load(FailureEnricherFactory.class);
+        final Set<FailureEnricher> failureEnrichers = new HashSet<>();
+        while (factoryIterator.hasNext()) {
+            try {
+                final FailureEnricherFactory failureEnricherFactory = factoryIterator.next();
+                final FailureEnricher failureEnricher =
+                        failureEnricherFactory.createFailureEnricher(configuration);
+                if (includedEnrichers.contains(failureEnricher.getClass().getName())) {
+                    failureEnrichers.add(failureEnricher);
+                    LOG.debug(
```

Comment:
This should be logged on the INFO level because that's the significant bit. We don't really care about the ones that were skipped (we can log those on DEBUG level)